sopel/plugins/handlers.py

```diff
@@ -618,8 +618,9 @@ def get_version(self) -> Optional[str]:
         ):
             try:
                 version = importlib.metadata.version(self.module.__package__)
-            except ValueError:
-                # package name is probably empty-string; just give up
+            except Exception:
+                # Just give up
+                # Can be caused by empty package name, mismatch between package/project names
```

Comment:
We've been talking about this one on IRC this evening, and I believe @SnoopJ and I have found a good middle ground:

If logging works here (we haven't tested how logging interacts with the CLI interfaces like `sopel-plugins`), it would be good to catch ``ValueError` and `PackageNotFoundError` which the `importlib.metadata.version()` call might raise, and just give up. Maybe log a warning, but that's it. As a fallback, catch `Exception` and log the full exception traceback for investigation. We'd like to know about other failure cases if they exist, but they shouldn't crash the CLI or the `.version` command called from IRC.

Reply:
Done in d5f4c1a. The logger does "work" on the command-line (i.e. `sopel-plugins list`) in the sense that it produces output. I think it happens because the logging hierarchy is unconfigured and Python falls back on assigning a default handler to the root of the hierarchy. At any rate, it doesn't fail silently which is the main thing we were after.